core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/RowDenormalizerAggregator.java

```diff
@@ -70,15 +72,20 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
 
   @Override
   public void initialize(BatchRuntimeContext context) throws Exception {
-    outputFields = conf.getOutputFields();
+    outputFields = conf.getOutputSchemaFields();
     outputMappings = conf.getFieldAliases();
     keyField = conf.getKeyField();
+    fieldName = conf.getFieldName();
+    fieldValue = conf.getFieldValue();
     outputSchema = initializeOutputSchema();
   }
 
   @Override
   public void groupBy(StructuredRecord record, Emitter<String> emitter) throws Exception {
     if (record.get(keyField) == null) {
+      if (!record.getSchema().getField(keyField).getSchema().isNullable()) {
```

Comment:
before doing this, you need to check that `record.getSchema().getField(keyField)` is not null. If it is, the key field does not exist in the input record.